core/controllers/acl_decorators_test.py

```diff
@@ -1632,6 +1634,150 @@ def test_viewer_cannot_accept_suggestion(self):
+    change_dict = {
+        'cmd': 'add_translation',
+        'content_id': 'content',
+        'language_code': 'hi',
+        'content_html': '<p>old content html</p>',
+        'state_name': 'State 1',
+        'translation_html': '<p>Translation for content.</p>'
+    }
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+        @acl_decorators.can_update_suggestions
+        def get(self, suggestion_id):
+            self.render_json({'suggestion_id': suggestion_id})
+
+    def setUp(self):
+        super(DecoratorForUpdatingSuggestionTests, self).setUp()
+        self.signup(self.author_email, self.author_username)
+        self.signup(self.user_email, self.username)
+        self.signup(self.owner_email, self.owner_username)
+        self.signup(self.reviewer_email, 'reviewer1')
+        self.author_id = self.get_user_id_from_email(self.author_email)
+        self.owner_id = self.get_user_id_from_email(self.owner_email)
+        self.reviewer_id = self.get_user_id_from_email(
+            self.reviewer_email)
+        self.owner = user_services.get_user_actions_info(self.owner_id)
+        self.author = user_services.get_user_actions_info(self.author_id)
+        user_services.update_user_role(self.owner_id, feconf.ROLE_ID_ADMIN)
+        user_services.allow_user_to_review_translation_in_language(
+            self.reviewer_id, 'hi')
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock/<suggestion_id>', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+
+        self.login(self.author_email)
+        exploration = (
+            self.save_new_linear_exp_with_state_names_and_interactions(
+                self.exploration_id, self.author_id, [
+                    'State 1', 'State 2', 'State 3'],
+                ['TextInput'], category='Algebra'))
+
+        self.old_content = state_domain.SubtitledHtml(
+            'content', '<p>old content html</p>').to_dict()
+        exploration.states['State 1'].update_content(
+            state_domain.SubtitledHtml.from_dict(self.old_content))
+        exploration.states['State 2'].update_content(
+            state_domain.SubtitledHtml.from_dict(self.old_content))
+        exploration.states['State 3'].update_content(
+            state_domain.SubtitledHtml.from_dict(self.old_content))
+        exp_services._save_exploration(self.author_id, exploration, '', [])  # pylint: disable=protected-access
+
+        rights_manager.publish_exploration(self.author, self.exploration_id)
+
+        self.new_content = state_domain.SubtitledHtml(
+            'content', '<p>new content html</p>').to_dict()
+        self.resubmit_change_content = state_domain.SubtitledHtml(
+            'content', '<p>resubmit change content html</p>').to_dict()
+
+        suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT, self.TARGET_TYPE,
```

Comment:
I suggest also adding a test to verify that question suggestions are not included with this decorator (e.g. that it will fail for questions despite all other needed criteria passing). This test should fail during the next PR once you add functionality (which is a nice spot-check that the new functionality is working), at which point it can be updated to verify that questions are supported rather than not.